src/extract.ts

```diff
@@ -13,8 +13,8 @@ export interface BenchmarkResult {
 
 interface GitHubUser {
     email?: string;
-    name: string;
```

Comment:
Why change these to optional?

Reply:
I made it optional because author and commiter that can be got from getCommit are nullable.
I will correct it if there is a better way to write it.

Reply:
Does it make more sense to make `author` and `committer` in `Commit` nullable? If this is not related to the main purpose in this PR, maybe we can remove it and have the discussion in a separated issue or PR.

Reply:
I am getting a build error, so I think I need to fix it somehow.

```
src/extract.ts:273:13 - error TS2322: Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.

273             name: commit.author?.name,
```

Using a non-null assertion operator instead of making it optional could fix the build error,
but I have not checked if this is a reasonable way to write it.

Reply:
I am getting a build error, so I think I need to fix it somehow.

```
src/extract.ts:273:13 - error TS2322: Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.

273             name: commit.author?.name,
```

Using a non-null assertion operator instead of making it optional could fix the build error,
but I have not checked if this is a reasonable way to write it.

https://github.com/benchmark-action/github-action-benchmark/pull/171#discussion_r1194581150

Reply:
Got it. The new version makes the returned field nullable. It looks good to me to also make it optional in our side.